mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -656,13 +656,15 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
         previousItemFactors.unpersist()
         itemFactors.setName(s"itemFactors-$iter").persist(intermediateRDDStorageLevel)
         // TODO: Generalize PeriodicGraphCheckpointer and use it here.
+        val deps = itemFactors.dependencies
```

Comment:
So it needs to be outside of the conditional because if I put it inside of the conditional it won't be in scope. I could make it an Option and capture the deps there but I think that would be a little convoluted.